src/main/java/frc/robot/libraries/PController.java

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.libraries;
+
+/** Add your docs here. */
+public class PController {
+
+    double kP = 0; // Const, non-final to allow setting
+    public void setP(double kP) {this.kP = kP;}
+
+    double target; // The value for the PController to approach
+    public void setTarget(double target) {this.target = target;}
+
+    double max, min; // Output clamp limits
+    public void setBoundaries(double max, double min) {
+        this.max = max; this.min = min;
+    }
+
+    /**
+     * A simple proportional controller implementation
+     * @param kP Proportion of the error
+     */
+    public PController(double kP, double minBoundary, double maxBoundary) {
+        this.kP = kP;
+        max = maxBoundary;
+        min = minBoundary;
+    }
+
+    /**
+     * Returns the proportion of the error from the target and
+     * given feedbackward value.
+     * @param feedBackward
+     * @return (feedBackward - target) * kP , Clamped to boundaries
+     */
+    public double calculate(double feedBackward) {
+        double deltaProportion = (feedBackward - target) * kP;
```

Comment:
this will get farther away proportionally faster. you need to do target - feedback